contracts/0.4.24/Lido.sol

```diff
@@ -79,33 +97,171 @@ contract Lido is ILido, IsContract, StETH, AragonApp {
+        uint256 _stakeLimitIncreasePerBlock
+    ) external {
+        _auth(STAKING_RESUME_ROLE);
+
+        (
+            uint256 maxStakeLimit,,
+            uint256 prevStakeLimit,
+        ) = STAKE_LIMIT_POSITION.getStorageUint256().decodeStakeLimitSlot();
+
+        // if staking was paused or unlimited previously,
+        // or new limit is lower than previous, then
+        // reset prev stake limit to max
+        if ((maxStakeLimit == 0) || (_maxStakeLimit < prevStakeLimit)) {
+            prevStakeLimit = _maxStakeLimit;
+        }
+
+        STAKE_LIMIT_POSITION.setStorageUint256(
+            StakeLimitUtils.encodeStakeLimitSlot(
+                _maxStakeLimit,
+                _stakeLimitIncreasePerBlock,
+                prevStakeLimit,
+                block.number
+            )
+        );
+
+        emit StakingResumed(_maxStakeLimit, _stakeLimitIncreasePerBlock);
+    }
+
+    /**
+    * @notice Check staking state: whether it's paused or not
+    */
+    function isStakingPaused() external view returns (bool) {
+        return STAKE_LIMIT_POSITION.getStorageUint256().isStakingPaused();
+    }
+
+    /**
+      * @notice Returns how much Ether can be staked in the current block
+      * @dev Special return values:
+      * - max uint256 if staking is unlimited;
+      * - 0 if staking is paused or if limit is exhausted.
+      */
+    function getCurrentStakeLimit() public view returns (uint256) {
+        uint256 slotValue = STAKE_LIMIT_POSITION.getStorageUint256();
+        if (!slotValue.isStakingRateLimited()) {
+            return uint256(-1);
+        }
+
+        return slotValue.calculateCurrentStakeLimit();
+    }
+
+    /**
+      * @notice Returns full info about stake limit
+      * @dev Might be used for the advanced integration requests.
+      * @return
+      * `isStakingPaused` the value returned by `isStakingPaused()`
+      * `isStakingLimitApplied` true if staking limit is set and false otherwise
+      * `currentStakeLimit` the value returned by `getCurrentStakingLimit()`
+      *
+      * Internals:
+      * `maxStakeLimit` internal max stake limit represenation
+      * `stakeLimitIncPerBlock` internal stake limit increase per block represenation
+      * `prevStakeLimit` internal previously reached stake limit represenation
+      * `prevStakeBlockNumber` internal prevously seen block number represenation
+      */
+    function getStakeLimitFullInfo() external view returns (
+        bool isStakingPaused,
+        bool isStakingLimitApplied,
+        uint256 currentStakeLimit,
+        uint256 maxStakeLimit,
+        uint256 stakeLimitIncPerBlock,
+        uint256 prevStakeLimit,
+        uint256 prevStakeBlockNumber
+    ) {
+        uint256 slotValue = STAKE_LIMIT_POSITION.getStorageUint256();
+        isStakingPaused = slotValue.isStakingPaused();
+        isStakingLimitApplied = slotValue.isStakingRateLimited();
+        currentStakeLimit = getCurrentStakeLimit();
```

Comment:
Maybe slotValue.calculateCurrentStakeLimit() to save some gas?

Reply:
Oh, I see for consistency let's do:
isStakingLimitApplied ? slotValue.calculateCurrentStakeLimit() : uint256(-1)